cabal-install/cabal-install.cabal

```diff
@@ -239,6 +245,11 @@ executable cabal
         zlib       >= 0.5.3    && < 0.7,
         hackage-security >= 0.5 && < 0.6
 
+    if flag(old-bytestring)
+      build-depends: bytestring <  0.10.2, bytestring-builder
+    else
+      build-depends: bytestring >= 0.10.2
```

Comment:
@dcoutts I'd just change it to `description:  Use bytestring < 0.10.2 and bytestring-builder`

@23Skidoo wouldn't hurt; fwiw, I stole the flag-logic from another packages (the `tar` package) which also had such a flag, and since we depend on `tar` this doesn't introduce any new dependency either